pkg/gardenlet/controller/shoot/care/health_test.go

```diff
@@ -460,6 +461,120 @@ var _ = Describe("health check", func() {
 		)
 	})
 
+	Describe("#CheckIfDependencyWatchdogProberScaledDownControllers", func() {
+		var (
+			deploymentCA  *appsv1.Deployment
+			deploymentKCM *appsv1.Deployment
+			deploymentMCM *appsv1.Deployment
+		)
+
+		BeforeEach(func() {
+			deploymentCA = &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: "cluster-autoscaler", Namespace: seedNamespace}, Spec: appsv1.DeploymentSpec{Replicas: ptr.To(int32(1))}}
+			deploymentKCM = &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: "kube-controller-manager", Namespace: seedNamespace}, Spec: appsv1.DeploymentSpec{Replicas: ptr.To(int32(1))}}
+			deploymentMCM = &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: "machine-controller-manager", Namespace: seedNamespace}, Spec: appsv1.DeploymentSpec{Replicas: ptr.To(int32(1))}}
+		})
+
+		It("should report nothing because no relevant deployment exists", func() {
+			scaledDownDeploymentNames, err := CheckIfDependencyWatchdogProberScaledDownControllers(ctx, fakeClient, seedNamespace)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(scaledDownDeploymentNames).To(BeEmpty())
+		})
+
+		It("should report nothing because all relevant deployment have replicas > 0", func() {
+			Expect(fakeClient.Create(ctx, deploymentCA)).To(Succeed())
+			Expect(fakeClient.Create(ctx, deploymentKCM)).To(Succeed())
+			Expect(fakeClient.Create(ctx, deploymentMCM)).To(Succeed())
+
+			scaledDownDeploymentNames, err := CheckIfDependencyWatchdogProberScaledDownControllers(ctx, fakeClient, seedNamespace)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(scaledDownDeploymentNames).To(BeEmpty())
+		})
+
+		It("should report names because some relevant deployment have replicas == 0", func() {
+			deploymentKCM.Spec.Replicas = nil
+			deploymentMCM.Spec.Replicas = ptr.To(int32(0))
+
+			Expect(fakeClient.Create(ctx, deploymentCA)).To(Succeed())
+			Expect(fakeClient.Create(ctx, deploymentKCM)).To(Succeed())
+			Expect(fakeClient.Create(ctx, deploymentMCM)).To(Succeed())
+
+			scaledDownDeploymentNames, err := CheckIfDependencyWatchdogProberScaledDownControllers(ctx, fakeClient, seedNamespace)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(scaledDownDeploymentNames).To(HaveExactElements(deploymentKCM.Name, deploymentMCM.Name))
+		})
+	})
+
+	Describe("#CheckForExpiredNodeLeases", func() {
+		var (
+			nodeName = "node1"
+			nodeList corev1.NodeList
+
+			validLease = &coordinationv1.Lease{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: nodeName,
+				},
```

Comment:
Not really important for the tested function, but we can include namespace in the lease object.